src/DeviceInterface/AppleDeviceInterface.js

```diff
@@ -304,34 +332,35 @@ class AppleDeviceInterface extends InterfacePrototype {
      *     - This also is triggered when the close event is called and prevents any edge case continued polling.
      * - 'ok' is when the user has selected a credential and the value can be injected into the page.
      * - 'none' is when the tooltip is open in the native window however hasn't been entered.
-     * @returns {Promise<{data:IdentityObject|CreditCardObject|CredentialsObject, configType: string} | null>}
+     * @param {(response: {data:IdentityObject|CreditCardObject|CredentialsObject, configType: string} | {stateChange: boolean} | {stop: boolean} | null) => void} callback
      */
-    async _listenForSelectedCredential () {
-        return new Promise((resolve) => {
-            // Prevent two timeouts from happening
-            // @ts-ignore
-            const poll = async () => {
-                clearTimeout(this.pollingTimeout)
-                const response = await this.getSelectedCredentials()
-                switch (response.type) {
-                case 'none':
-                    // Parent hasn't got a selected credential yet
-                    // @ts-ignore
-                    this.pollingTimeout = setTimeout(() => {
-                        poll()
-                    }, 100)
-                    return
-                case 'ok': {
-                    return resolve({data: response.data, configType: response.configType})
-                }
-                case 'stop':
-                    // Parent wants us to stop polling
-                    resolve(null)
-                    break
-                }
+    async _listenForSelectedCredential (callback) {
+        // Prevent two timeouts from happening
+        const poll = async () => {
+            clearTimeout(this.pollingTimeout)
+            const response = await this.getSelectedCredentials()
+            switch (response.type) {
+            case 'none':
+                // Parent hasn't got a selected credential yet
+                this.pollingTimeout = setTimeout(() => poll(), 100)
+                return
+            case 'ok': {
+                await callback({data: response.data, configType: response.configType})
+                return
             }
-            poll()
-        })
+            case 'state': {
+                // Inform that state has changed, but continue polling
+                // e.g. in-context signup has been dismissed
```

Comment:
Why should we keep polling if in-context has been dismissed?

Reply:
@GioSensation When there are identities available, dismissing in-context signup doesn't close the tooltip, it just removes the in-context signup option. We'll want to keep polling so that selecting the identity after dismissing works as expected.